inception-project/inception
# #3113 - Option to wait for recommendations before showing document

**What's in the PR**
- Added a per-project option to wait for non-trainable recommenders to provide suggestions before accessing a document (off by default)
- Refactoring of the recommender tasks
- Refactoring of the selection task to be more readable

**How to test manually**
* Configure a non-trainable recommender, e.g. a non-trainable external recommender or an ELG recommender
* Access the annotation page with the option disabled - access should be immediate, recommendations may appear only later
* Log out - log back in
* Enable the wait option and then access the annotation page - access should be blocked until recommendations have been retrieved (or until ELG runs into a timeout...)

**Automatic testing**

**Documentation**
